packages/cli/src/util/deploy/process-deployment.ts

```diff
@@ -281,3 +285,30 @@ export default async function processDeployment({
     throw err;
   }
 }
+
+export const archiveSuggestionText =
+  'Try using `--archive=tgz` to limit the amount of files you upload.';
+
+export class UploadErrorMissingArchive extends Error {
+  link = 'https://vercel.com/docs/cli/deploy#archive';
+}
+
+export function handleErrorSolvableWithArchive(error: unknown) {
+  const errorIsObject = typeof error === 'object' && error !== null;
```

Comment:
There's probably a helper or two in `@vercel/error-utils` that would be useful here.